### ISSUE TITLE
With the version 0.24.1 it is not possible to remove the affinity host on an existing VM.

### ISSUE THREAD

**Issue by bufanda** [terra-farm/terraform-provider-xenorchestra#238]
I have a VM I had configured an affinity host, now with an expanded cluster and reworking some of the VM distribution I want this one particular VM to not have a affinity to one particular host anymore. So I removed the affinity host rule from the terraform file but now I get the following error.

```
╷
│ Error: jsonrpc2: code 10 message: invalid parameters: {"errors":[{"instancePath":"/affinityHost","schemaPath":"#/properties/affinityHost/minLength","keyword":"minLength","params":{"limit":1},"message":"must NOT have fewer than 1 characters"}]}
│
│   with xenorchestra_vm.aegir,
│   on instance-aegir.tf line 1, in resource "xenorchestra_vm" "aegir":
│    1: resource "xenorchestra_vm" "aegir" {
│
╵
```

As the affinity host rule os only optional per documentation shouldn't it be possible to remove the rule and have terraform remove the setting from the host? Or do I have to destroy the VM and recreate it. Which sure would be possible but it isn't the behaviour I would expect. As in the gui I can simply click on the 'X' to remove the affinity.

**Comment by bufanda** [comment]
Hi,

the ignore_changes works for now. Thanks.

**Comment by RealStickman** [comment]
I found a workaround to remove affinity. Simply set the `affinity_host` the pool id. This makes the affinity `None` in Xen Orchestra.  
```
  affinity_host = data.xenorchestra_pool.pool.id
```

**Comment by bufanda** [comment]
> I found a workaround to remove affinity. Simply set the `affinity_host` the pool id. This makes the affinity `None` in Xen Orchestra.
> 
> ```
>   affinity_host = data.xenorchestra_pool.pool.id
> ```

Tried it. When you have a VM stopped it won't start anymore as the affinity host wiht the pool id doesn't exists. 

You'll get an error message like this
```
You gave an invalid object reference. The object may have recently been deleted. The class parameter gives the type of reference given, and the handle parameter echoes the bad value given.
class: host
handle: b0b45af2-b561-5cc7-fc28-0a65b3aa32e0
```

**Comment by RealStickman** [comment]
@bufanda yeah, you're right. I should've tested that more than just checking if the host was removed.

**Comment by ddelnano** [comment]
This will be fixed in v0.24.2, which will be available when [this](https://github.com/terra-farm/terraform-provider-xenorchestra/actions/runs/5864556327) GitHub action completes.
